### ISSUE TITLE
auto update lockscreen

### ISSUE THREAD

**Issue by foreteller22** [liaoheng/BingWallpaper#3]
Hi
Auto update does not work on Nokia 3.1 ,Oreo 8.0 . App is of course taken out from battery optimization. I need to start app and then picture is updating.

**Comment by liaoheng** [comment]
Open the feedback dialog and click the Copy button to send the content to me, or you can try adding a desktop widget.

**Comment by foreteller22** [comment]
Hi

feedback info -------------------------
 sdk: 26 device: ES2N_sprout model: Nokia 3.1 product: Essential2_00WW
rom_name: HMD Global rom_version: 00WW_0_46E locale: sk_SK auto_locale:
sk_SK version: 1.2.4 job: google_service alarm: false alarm_time:
feedback info -------------------------


And I do not understand. I have to create widget in spite lockscreen
getting updated ?

Thanks you

Dňa po 20. 8. 2018, 4:02 Leng <notifications@github.com> napísal(a):

**Comment by liaoheng** [comment]
1. You desktop wallpaper update works fine ? or only lockscreen is failure. If the former is not normal,  Please follow the [note](https://github.com/liaoheng/BingWallpaper/blob/image/HELP.md#feedback) and sending to me. 
2. add a desktop widget to help the app work Instead of getting updates.

**Comment by foreteller22** [comment]
Hi
1. I do not use app for wallpaper . I only use it for lockscreen . Far wallpaper I use still one picture.
2. I added widget today ( 21.8 ) so I will see.
Debug log is attached.
[debug_log.txt](https://github.com/liaoheng/BingWallpaper/files/2304874/debug_log.txt)

**Comment by foreteller22** [comment]
1. Daily automatic update. ( But strange is : In app it is set to daily automatic update. But when I add widget and start app from widget there was this setting off , all of them. So I set it there too ).
I will check logs , and let you know it is solved my problem.
Thank you

**Comment by foreteller22** [comment]
So after adding widget lockscreen update still does not work. I am attaching log. But it is maybe caused by Nokia. So I going to wait for firmware update and will see then. 
[debug_log.txt](https://github.com/liaoheng/BingWallpaper/files/2313144/debug_log.txt)

**Comment by liaoheng** [comment]
See log it is work fine, try setting option: 
 ![snipaste_2018-08-23_16-39-16](https://user-images.githubusercontent.com/10915809/44514621-2999a900-a6f3-11e8-936c-d67908555ff5.png)
![snipaste_2018-08-23_16-39-30](https://user-images.githubusercontent.com/10915809/44514674-564dc080-a6f3-11e8-91ae-fc4beb505df5.png)

**Comment by foreteller22** [comment]
No , it is not works properly. When you see logs : in time 11:59:53.625  - this was auto update , after restart of device . Auto update has timing every 3 hours . Update in time 15:02:10.830 was made manualy , when I was waiting if till 15:01. Actually of this device ( Nokia 3.1 with Android 8.0 ) almost all background tasks works only once after restart of device and then stops. And thanks for your hint , but I already got it set as you show. And off course I got your app excluded from battery optimization.
